spanner/opentelemetry/src/main/java/com/example/spanner/CaptureSpannerMetric.java

```diff
@@ -0,0 +1,312 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.spanner;
+
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.ReadContext.QueryAnalyzeMode;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spanner.Statement;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.ListeningScheduledExecutorService;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.google.protobuf.Value;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.metrics.DoubleHistogram;
+import io.opentelemetry.api.metrics.Meter;
+import io.opentelemetry.exporter.otlp.metrics.OtlpGrpcMetricExporter;
+import io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.metrics.SdkMeterProvider;
+import io.opentelemetry.sdk.metrics.export.PeriodicMetricReader;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.export.SimpleSpanProcessor;
+import io.opentelemetry.sdk.trace.samplers.Sampler;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.Executors;
+
+/**
+ * This sample demonstrates how to capture Cloud Spanner's Query Stats latency using OpenTelemetry.
+ */
+public class CaptureSpannerMetric {
+
+  // [START spanner_opentelemetry_usage]
+  public static void main(String[] args) throws Exception {
+    // TODO(developer): Replace these variables before running the sample.
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "span-cloud-testing";
+    String instanceId = "surbhi-testing";
+    String databaseId = "test-db";
+
+    // Enable OpenTelemetry metrics and traces before Injecting OpenTelemetry
+    SpannerOptions.enableOpenTelemetryMetrics();
+    SpannerOptions.enableOpenTelemetryTraces();
+
+    OpenTelemetry openTelemetry = openTelemetry();
+
+    SpannerOptions options = SpannerOptions.newBuilder()
+        .setOpenTelemetry(openTelemetry)
+        .build();
+    Spanner spanner = options.getService();
+    DatabaseClient dbClient = spanner
+        .getDatabaseClient(DatabaseId.of(projectId, instanceId, databaseId));
+
+    captureGfeMetric(dbClient);
+    registerQueryStatsMetric(openTelemetry);
```

Comment:
Let's move this to before SpannerOptions and immediately after openTelemetry() call given that this is part of the setup and is called at-most once.